cli/src/build-settings.ts

```diff
@@ -54,25 +65,32 @@ export default class BuildSettings {
 
   static fromJSON(value: JSON.Value): BuildSettings {
     if (!JSON.isObject(value)) {
-      throw new TypeError("value is not an object");
+      throw new TypeError("value must be an object");
     }
-    let rustc = value.rustc;
-    let env = value.env;
+    let { rustc, env, nodeVersion } = value;
     if (typeof rustc !== 'string') {
-      throw new TypeError("value.rustc is not a string");
+      throw new TypeError("value.rustc must be a string");
+    }
+    if (!('nodeVersion' in value)) {
```

Comment:
I think this is not the safest logic: the case where there's no `'nodeVersion'` key is when they used a previous version of Neon, in which case we don't know what Node version they were using, so we should leave it `null` to allow ourselves to cache-bust.